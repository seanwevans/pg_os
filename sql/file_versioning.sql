-- tests for file versioning on repeated writes
\set ECHO none
SET client_min_messages TO warning;
DROP EXTENSION IF EXISTS pg_os CASCADE;
CREATE EXTENSION pg_os;
\set ECHO queries
\set VERBOSITY terse

-- setup: create writer user with file write permission
SELECT create_user('writer') AS writer_id \gset
SELECT create_role('file_writer') AS role_id \gset
SELECT assign_role_to_user(:writer_id, :role_id);
SELECT grant_permission_to_role(:role_id, 'file', 'write');

-- create a file owned by the writer
SELECT create_file(:writer_id, 'versioned.txt', NULL, FALSE) AS file_id \gset

-- write twice, generating two versions
SELECT write_file(:writer_id, :file_id, 'first revision');
SELECT write_file(:writer_id, :file_id, 'second revision');

-- ensure versions capture each prior state
SELECT file_id, version_number, contents
  FROM file_versions
 WHERE file_id = :file_id
 ORDER BY version_number;
