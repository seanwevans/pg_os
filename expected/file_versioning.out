-- tests for file versioning on repeated writes
\set ECHO none
SELECT create_user('writer') AS writer_id
SELECT create_role('file_writer') AS role_id
SELECT assign_role_to_user(1, 1);
 assign_role_to_user
---------------------

(1 row)

SELECT grant_permission_to_role(1, 'file', 'write');
 grant_permission_to_role
--------------------------

(1 row)

SELECT create_file(1, 'versioned.txt', NULL, FALSE) AS file_id
 file_id
---------
       1
(1 row)

SELECT write_file(1, 1, 'first revision');
 write_file
------------

(1 row)

SELECT write_file(1, 1, 'second revision');
 write_file
------------

(1 row)

SELECT file_id, version_number, contents
  FROM file_versions
 WHERE file_id = 1
 ORDER BY version_number;
 file_id | version_number |     contents
---------+----------------+-----------------
       1 |              1 | 
       1 |              2 | first revision
(2 rows)

